src/main/java/net/minecraftforge/client/model/animation/Clips.java

```diff
@@ -0,0 +1,236 @@
+package net.minecraftforge.client.model.animation;
+
+import java.io.IOException;
+
+import net.minecraft.util.MathHelper;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.model.IModel;
+import net.minecraftforge.client.model.IModelPart;
+import net.minecraftforge.client.model.IModelState;
+import net.minecraftforge.client.model.ModelLoaderRegistry;
+import net.minecraftforge.client.model.TRSRTransformation;
+import net.minecraftforge.fml.common.FMLLog;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+
+/**
+ * Various implementations of IClip, and utility methods.
+ */
+public class Clips
+{
+    /**
+     * Clip that does nothing.
+     */
+    public static enum IdentityClip implements IClip
+    {
+        instance;
```

Comment:
Forge uses both instance and INSTANCE; lowercase is in my opinion less jarring, so I used that.